doc/shield-coinbase.md

```diff
@@ -0,0 +1,101 @@
+- remainingUTXOs: number of coinbase utxos still available for shielding.
+- remainingValue: value of coinbase utxos still available for shielding
+
+### Locking UTXOs
+
+The `z_shieldcoinbase` call will lock any selected UTXOs. This prevents the selected UTXOs which are already queued up from being selected for any other send operation.  If the `z_shieldcoinbase` call fails, any locked UTXOs are unlocked.
+
+You can use the RPC call `lockunspent` to see which UTXOs have been locked.  You can also use this call to unlock any UTXOs in the event of an unexpected system failure which leaves UTXOs in a locked state.
+
+### Limits, Performance and Transaction Confirmation
+
+The number of coinbase utxos selected for shielding can be adjusted by setting the limit parameter. The default value is 50.
+
+If the limit parameter is set to zero, the zcashd `mempooltxinputlimit` option will be used instead, where the default value for `mempooltxinputlimit` is zero which means no limit.
+
+Any limit is constrained by a hard limit due to the consensus rule defining a maximum transaction size of 100,000 bytes.
+
+In general, the more utxos that are selected, the longer it takes for the transaction to be verified.  Due to the quadratic hashing problem, some miners use the `mempooltxinputlimit` option to reject transactions with a large number of UTXO inputs.
+
+Currently, as of November 2017, there is no commonly agreed upon limit, but as a rule of thumb (a form of emergent consensus) if a transacion has less than 100 UTXO inputs, the transaction will be mined promptly by the majority of mining pools, but if it has say 500 UTXO inputs it might take several days to be mined by a miner who has higher or no limits.
+
+### Anatomy of a z_shieldcoinbase transaction
+
+The transaction created is a shielded transaction.  It consists of a single joinsplit, which consumes coinbase UTXOs as input, and deposits value at a shielded address, minus any fee.
+
+The number of coinbase UTXOs is determined by a user configured limit.
+
+If no limit is set (set limit parameter and `mempooltxinputlimit` option both to zero) the behaviour of z_shieldcoinbase is consume as many utxos as possible, with `z_shieldcoinbase` constructing a transaction upto the size limit of 100,000 bytes.
+
+As a result, the maximum numper of inputs that can be selected is
+
+- P2PKH coinbase utxos ~ 662
+- 2-of-3 multisig P2SH coinbase utxos ~ 244.
+
+Here is an example of using `z_shieldcoinbase` on testnet to shield multi-sig coinbase utxos.
+
+- Block 141042 is almost ~2 MB in size (the maximum size for a block) and contains 1 coinbase reward transaction and 20 transactions, each indivually created by a call to z_shieldcoinbase.
+  - https://explorer.testnet.z.cash/block/0050552a78e97c89f666713c8448d49ad1d7263274422272696187dedf6c0d03
+- Drilling down into a transaction, you can see there is one joinsplit, with 244 inputs (vin) inputs, an no outputs (0 vout).
```

Comment:
word "inputs" is repeated here: "244 inputs (vin) inputs"
s/an no/and no

Reply:
ok